haystack/components/preprocessors/recursive_splitter.py

```diff
@@ -88,12 +93,19 @@ def __init__(
         self.sentence_splitter_params = (
             {"keep_white_spaces": True} if sentence_splitter_params is None else sentence_splitter_params
         )
+        self.tiktoken_tokenizer: Optional["tiktoken.Encoding"] = None
+        self._is_warmed_up = False
```

Comment:
👍🏽 this is a good approach we could adopt this as the default for every component that needs being `warmed_up`-  so that all have the same approach and not different ones

Reply:
I took it from the Agent :)
There is an option issue for handling warm up more consistently: https://github.com/deepset-ai/haystack/issues/8823